src/app/(sidebar)/smart-contracts/contract-explorer/components/ContractSpec.tsx

```diff
@@ -37,6 +47,47 @@ export const ContractSpec = ({
     isActive: Boolean(isActive && rpcUrl && wasmHash),
   });
 
+  const {
+    data: wasmBinary,
+    error: wasmBinaryError,
+    isLoading: isWasmBinaryLoading,
+    isFetching: isWasmBinaryFetching,
+    refetch: fetchWasmBinary,
+  } = useSEContractWasmBinary({ wasmHash, networkId });
+
+  const resetWasmBlob = useCallback(() => {
+    queryClient.resetQueries({
+      queryKey: ["useSEContractWasmBinary", networkId, wasmHash],
+    });
+  }, [networkId, queryClient, wasmHash]);
+
+  useEffect(() => {
```

Comment:
can we do this onClick instead of lifecycle?

Reply:
The click triggers React Query to download the wasm binary, so we need to do something after this step happens. The other option would be to put this inside the React Query, but that would limit the usage of that query to just this.

I'm not a huge fan of this approach, but I couldn't come up with something better. I'm open to suggestions! :pray: